src/blockchain/Blockchain.ts

```diff
@@ -27,19 +27,70 @@ export type Stats = {
   nodeVersion: string
 }
 
+export type txStatusPromiseResolver = (result: SubmittableResult) => boolean
+
 export interface IBlockchainApi {
   api: ApiPromise
   portablegabi: gabi.Blockchain
 
   getStats(): Promise<Stats>
   listenToBlocks(listener: (header: Header) => void): Promise<() => void>
-  submitTx(
+  signTx(
     identity: Identity,
     tx: SubmittableExtrinsic
+  ): Promise<SubmittableExtrinsic>
+  submitTx(
+    identity: Identity,
+    tx: SubmittableExtrinsic,
+    resolveOn?: txStatusPromiseResolver
   ): Promise<SubmittableResult>
   getNonce(accountAddress: string): Promise<Codec>
 }
 
+export const AWAIT_READY: txStatusPromiseResolver = (result) =>
+  result.status.isReady
+export const AWAIT_IN_BLOCK: txStatusPromiseResolver = (result) =>
+  result.isInBlock
+export const AWAIT_FINALIZED: txStatusPromiseResolver = (result) =>
+  result.isFinalized
+
+export async function submitTx(
+  tx: SubmittableExtrinsic,
+  resolveOn: txStatusPromiseResolver = AWAIT_FINALIZED
+): Promise<SubmittableResult> {
+  log.info(`Submitting ${tx.method}`)
+  let unsubscribe: () => void
+  return new Promise<SubmittableResult>((resolve, reject) => {
+    tx.send((result) => {
+      log.info(`Got tx status '${result.status.type}'`)
+      if (ErrorHandler.extrinsicFailed(result)) {
+        log.warn(`Extrinsic execution failed`)
+        log.debug(`Transaction detail: ${JSON.stringify(result, null, 2)}`)
+        const extrinsicError: ExtrinsicError =
+          ErrorHandler.getExtrinsicError(result) || ERROR_UNKNOWN
+        log.warn(`Extrinsic error ocurred: ${extrinsicError}`)
+        reject(extrinsicError)
+      } else if (result.isError) {
+        reject(
+          new Error(`Transaction failed with status '${result.status.type}'`)
+        )
+      } else if (resolveOn(result)) {
+        resolve(result)
+      }
+    })
+      .then((cb) => {
+        unsubscribe = cb
+      })
+      // not sure we need this final catch block
+      .catch((err: Error) => {
+        // just reject with the original tx error from the chain
+        reject(err)
```

Comment:
What errors are you trying to catch with this?

Reply:
I don't know... that's old code, I added the comment `not sure we need this final catch block`

Reply:
we actually have failing tests if that's not there, so I guess we need it

Reply:
Ok, that is fine.  I was just wondering as well. It seemed like everything was handled. 

Do you know if there is a better way to catch if no tokens are available.

Reply:
This isn't a big deal, maybe we would like to make a ticket to see if we can have a better way of handling this.

Reply:
it's just weird for me as I expected anything that throws inside of a promise to be implicitly handled and to reject the promise, but it seems we have to make it explicit here